acceptance/tests/resource/user/should_create_with_expiry.rb

```diff
@@ -0,0 +1,30 @@
+test_name "verifies that puppet resource creates a user and assigns the correct expiry date when absent"
+confine :except, :platform => 'windows'
+confine :except, :platform => /^eos-/ # See ARISTA-37
+confine :except, :platform => /^cisco_/ # See PUP-5828
+
+tag 'audit:high',
+    'audit:refactor',  # Use block style `test_run`
+    'audit:acceptance' # Could be done as integration tests, but would
+                       # require changing the system running the test
+                       # in ways that might require special permissions
+                       # or be harmful to the system running the test
+
+user = "pl#{rand(999999).to_i}"
+
+agents.each do |host|
+  step "user should not exist"
+  host.user_absent(user)
```

Comment:
this seems to fail on OS X

```
if dscl . -list /Users/pl872282; then dscl . -delete /Users/pl872282; fi
<dscl_cmd> DS Error: -14009 (eDSUnknownNodeName)
list: Invalid Path
```

Can we use `on(host, puppet_resource('user', user, 'ensure=absent'))` ? that seems to work
```
puppet resource user pl872282 ensure=absent
Notice: /User[pl872282]/ensure: removed
user { 'pl872282':
   ensure => 'absent',
}
```